NYPLAudiobookToolkit/Network/AudiobookNetworkService.swift

```diff
@@ -139,6 +169,7 @@ extension DefaultAudiobookNetworkService: DownloadTaskDelegate {
 
     public func downloadTaskFailed(_ downloadTask: DownloadTask, withError error: NSError?) {
         self.cursor = nil
+        self.isDownloading = false
```

Comment:
do we not need to lock before modifying the flag here? I'm not sure if this is a special case compared to the other methods, but at first glance it seems like this can compromise the integrity of the flag state if we don't lock.